internal/async/goroutine.go

```diff
@@ -13,3 +22,41 @@ func init() {
 func IsMainGoroutine() bool {
 	return goroutineID() == mainGoroutineID
 }
+
+func EnsureNotMain(fn func()) {
+	if build.DisableThreadChecks || !IsMainGoroutine() {
+		fn()
+		return
+	}
+
+	log.Println("*** Error in Fyne call thread, fyne.Do called from main goroutine ***")
+
+	logStackTop(2)
+	go fn()
+}
+
+func EnsureMain(fn func()) {
```

Comment:
I kind of feel like these should have comments as they are exported (although in an internal package but anyway)? Applies to the ones above as well.